test/local/mysql_tests.js

```diff
@@ -42,20 +42,51 @@ describe('MySQL', () => {
   )
 
   it(
-    'forces REQUIRED_SQL_MODES for connections',
+    'accepts REQUIRED_SQL_MODES from config',
     () => {
-      const configSqlMode = Object.assign({}, config)
-      const REQUIRED_SQL_MODES = [
-        'STRICT_ALL_TABLES',
-        'NO_ENGINE_SUBSTITUTION',
-      ]
-      var mode = REQUIRED_SQL_MODES.join(',')
-      configSqlMode.sql_mode = 'xyz'
-      return DB.connect(configSqlMode)
+      const configModes = Object.assign({}, config)
+      configModes.requiredSQLModes = 'STRICT_TRANS_TABLES,NO_ZERO_DATE'
+
+      return DB.connect(configModes)
+        .then(
+          db => {
+            assert.deepEqual(db.requiredModes, [
+              'STRICT_TRANS_TABLES',
+              'NO_ZERO_DATE'
+            ])
+          },
+          assert.fail
+        )
+    }
+  )
+
+  it(
+    'rejects unrecognized REQUIRED_SQL_MODES values from config',
+    () => {
+      const configModes = Object.assign({}, config)
+      configModes.requiredSQLModes = 'UNRECOGNIZED_SQL_MODE_NONSENSE'
+
+      return DB.connect(configModes)
+        .then(
+          assert.fail,
+          err => {
+            assert.equal(err.message, 'ER_WRONG_VALUE_FOR_VAR')
```

Comment:
The fact that MySQL rejects this value, is a useful test that we are actually attempting to call `SET SESSION sql_mode = blah` with the provided value.